pep-0659.rst

```diff
@@ -99,47 +99,43 @@ more robust in cases where specialization fails or is not stable.
 Performance
 -----------
 
-The expected speedup of 50% can be broken roughly down as follows:
+The speedup from specialization is hard to determine, as many specializations
+depend on other optimizations. Speedups seem to be in the range 10% - 60%.
 
-* In the region of 30% from specialization. Much of that is from
-  specialization of calls, with improvements in instructions that are already
-  specialized such as ``LOAD_ATTR`` and ``LOAD_GLOBAL`` contributing much of
-  the remainder. Specialization of operations adds a small amount.
-* About 10% from improved dispatch such as super-instructions
-  and other optimizations enabled by quickening.
-* Further increases in the benefits of other optimizations,
-  as they can exploit, or be exploited by specialization.
+* Most of the speedup comes directly from specialization. The largest
+  contributors are speedups to attribute lookup, global variables, and calls.
+* A small, but useful, fraction is from from improved dispatch such as
+  super-instructions and other optimizations enabled by quickening.
 
 Implementation
 ==============
 
 Overview
 --------
 
-Once any instruction in a code object has executed a few times,
-that code object will be "quickened" by allocating a new array for the
-bytecode that can be modified at runtime, and is not constrained as the
-``code.co_code`` object is. From that point onwards, whenever any
-instruction in that code object is executed, it will use the quickened form.
-
 Any instruction that would benefit from specialization will be replaced by an
```

Comment:
Any reason why you don't discuss the warmup period anymore? Perhaps just because we're moving towards quickening everything now?

Reply:
Yes, quickening everything, means that "warm up" is handled by the adaptive instructions.